From 1a83d13e2fca65022f5b16b3b2d5fea572b5ad3b Mon Sep 17 00:00:00 2001
From: Ruchika Gupta <ruchika.gupta@freescale.com>
Date: Mon, 22 Apr 2013 11:50:04 +0530
Subject: [PATCH 179/430] crypto: caam - Packed RNG vid function

Packed RNG vid function into caam_probe function

Signed-off-by: Ruchika Gupta <ruchika.gupta@freescale.com>
Change-Id: I3fb28ee0ae4daec89725868900445d36fe7e8795
Reviewed-on: http://git.am.freescale.net:8181/1675
Reviewed-by: Garg Vakul-B16394 <vakul@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
[original patch is from QorIQ-SDK-V1.4-20130625-yocto]
Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 drivers/crypto/caam/ctrl.c |   16 ++++------------
 1 files changed, 4 insertions(+), 12 deletions(-)

diff --git a/drivers/crypto/caam/ctrl.c b/drivers/crypto/caam/ctrl.c
index 07a13e2..019511f 100644
--- a/drivers/crypto/caam/ctrl.c
+++ b/drivers/crypto/caam/ctrl.c
@@ -190,17 +190,6 @@ int caam_get_era(u64 caam_id)
 }
 EXPORT_SYMBOL(caam_get_era);
 
-/**
- * get_rng_vid() - Return the version of the (T)RNG  block of the SEC.
- * @topregs - pointer to SEC register space
- **/
-static inline u8 get_rng_vid(struct caam_full __iomem *topregs)
-{
-	u64 cha_vid = rd_reg64(&topregs->ctrl.perfmon.cha_id);
-
-	return (cha_vid & CHA_ID_RNG_MASK) >> CHA_ID_RNG_SHIFT;
-}
-
 /* Probe routine for CAAM top (controller) level */
 static int caam_probe(struct platform_device *pdev)
 {
@@ -214,6 +203,7 @@ static int caam_probe(struct platform_device *pdev)
 #ifdef CONFIG_DEBUG_FS
 	struct caam_perfmon *perfmon;
 #endif
+	u64 cha_vid;
 
 	ctrlpriv = kzalloc(sizeof(struct caam_drv_private), GFP_KERNEL);
 	if (!ctrlpriv)
@@ -309,7 +299,9 @@ static int caam_probe(struct platform_device *pdev)
 	 * RNG4 based SECs need special initialization prior
 	 * to executing any descriptors
 	 */
-	if (get_rng_vid(topregs) >= 4) {
+	cha_vid = rd_reg64(&topregs->ctrl.perfmon.cha_id);
+
+	if ((cha_vid & CHA_ID_RNG_MASK) >> CHA_ID_RNG_SHIFT >= 4) {
 		kick_trng(pdev);
 		ret = instantiate_rng(ctrlpriv->jrdev[0]);
 		if (ret) {
-- 
1.7.5.4

